lib/inspec/input.rb

```diff
@@ -318,6 +318,21 @@ def has_value?
       !current_value.is_a? NO_VALUE_SET
     end
 
+    #--------------------------------------------------------------------------#
+    #                               Marshalling
+    #--------------------------------------------------------------------------#
+
+    def to_hash
+      as_hash = { name: name, options: {} }
+      %i{description title identifier type required value}.each do |field|
+        val = send(field)
+        next if val.nil?
+
+        as_hash[:options][field] = val
+      end
+      as_hash
+    end
```

Comment:
This code is duplicated from https://github.com/inspec/inspec/blob/master/lib/inspec/objects/input.rb#L17-L26 . It should only live in one spot, esp since those two files open and define in the same class (which is another separate problem).

Why it isn't showing up in codeclimate is probably a threshold thing and this method is juuuust small enough.

Reply:
Well, I've modified it to `require "inspec/objects/input"`. Is that a lesser evil?

Reply:
It is A lesser evil... I'm not sure it is the right one.

My intent is to move `lib/inspec/objects/*` to inspec-scap. I know we use `to_hash` in inspec, but I don't see use in scap (there are 3 calls to to_hash in `or_test.rb` and `control.rb`, but I don't think they're really necessary).

I don't think the other to_hash methods are actually needed, but I'm double checking that now.

Reply:
In discussion Friday October 4th, @zenspider felt that the least evil at this point would be to revert the code to have a duplicate `to_hash` implementation in both locations, and remove the `require "inspec/objects/input"`. I'll push that shortly.